From 766bbb22503948c4d37b28e46850ffdf41da5725 Mon Sep 17 00:00:00 2001
From: Amine Kristou <aminex.kristou@intel.com>
Date: Tue, 14 Oct 2014 19:00:32 +0200
Subject: [PATCH] tinyalsa: add pcm_get_subdevice()

The users of tinyalsa had no way of knowing on which subdevice a stream had
been created.  A new API, "unsigned int pcm_get_subdevice(struct *pcm)" returns
it.  This information is filled during the pcm_open()

Change-Id: I4297c3a3e3991e70617ab07fffa4ac92ca70c4f9
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-4097
Category: aosp improvement
Domain: AudioComms-Tinyalsa
Origin: internal
Upstream-Candidate: yes
Signed-off-by: Amine Kristou <aminex.kristou@intel.com>
---
 include/tinyalsa/asoundlib.h |    3 +++
 pcm.c                        |    7 +++++++
 2 files changed, 10 insertions(+)

diff --git a/include/tinyalsa/asoundlib.h b/include/tinyalsa/asoundlib.h
index e87d1aa..49524ed 100644
--- a/include/tinyalsa/asoundlib.h
+++ b/include/tinyalsa/asoundlib.h
@@ -217,6 +217,9 @@ unsigned int pcm_get_latency(struct pcm *pcm);
 int pcm_get_htimestamp(struct pcm *pcm, unsigned int *avail,
                        struct timespec *tstamp);
 
+/* Returns the subdevice on which the pcm has been opened */
+unsigned int pcm_get_subdevice(struct pcm *pcm);
+
 /* Write data to the fifo.
  * Will start playback on the first write or on a write that
  * occurs after a fifo underrun.
diff --git a/pcm.c b/pcm.c
index ff151fb..7fc829e 100644
--- a/pcm.c
+++ b/pcm.c
@@ -258,6 +258,7 @@ struct pcm {
     unsigned int noirq_frames_per_msec;
     int wait_for_avail_min;
     long pcm_delay;
+    unsigned int subdevice;
 };
 
 unsigned int pcm_get_buffer_size(struct pcm *pcm)
@@ -270,6 +271,11 @@ const char* pcm_get_error(struct pcm *pcm)
     return pcm->error;
 }
 
+unsigned int pcm_get_subdevice(struct pcm *pcm)
+{
+    return pcm->subdevice;
+}
+
 static int oops(struct pcm *pcm, int e, const char *fmt, ...)
 {
     va_list ap;
@@ -867,6 +873,7 @@ struct pcm *pcm_open(unsigned int card, unsigned int device,
         oops(pcm, errno, "cannot get info");
         goto fail_close;
     }
+    pcm->subdevice = info.subdevice;
 
     param_init(&params);
     param_set_mask(&params, SNDRV_PCM_HW_PARAM_FORMAT,
-- 
1.7.9.5

